### PR TITLE
allow a player to store their pronouns as part of their profile

### PR DESCRIPTION
**Pull request by jaburx**
Adds pronouns as a profile characteristic.

Paired With: @vidvidvid

### CODE REVIEW

**Comment by dan13ram** [comment]
@jaburx Thank you so much for working on this.

Could you take a look at the build issues ? Seems like typecheck is failing.

Also would be super great if you spit this PR into two where the first one includes *only* backend changes, and while the second one include *only* frontend changes. 

This would greatly help in the code review & merge process, and also that way the frontend only PR can have a working preview deployment to test with :) 

Cheers!

File: packages/web/components/Player/Section/PlayerPronouns.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { MetaTag } from '@metafam/ds';
+import { PlayerFragmentFragment } from 'graphql/autogen/types';
+import React from 'react';
+
+type PlayerPronounsProps = {
+  player: PlayerFragmentFragment;
+};
+
+export const PlayerPronouns = ({ player }: PlayerPronounsProps) => (
+  <>
```
Comment (dysbulic) [inline packages/web/components/Player/Section/PlayerPronouns.tsx:10]:
```suggestion
```

File: packages/web/components/Player/Section/PlayerPronouns.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    {player.pronouns && (
```
Comment (dysbulic) [inline packages/web/components/Player/Section/PlayerPronouns.tsx:11]:
```suggestion
  player.pronouns && (
```

File: packages/web/components/Player/Section/PlayerPronouns.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      <MetaTag size="md" fontWeight="normal" backgroundColor="gray.600">
+        {player.pronouns}
+      </MetaTag>
+    )}
```
Comment (dysbulic) [inline packages/web/components/Player/Section/PlayerPronouns.tsx:15]:
```suggestion
    )
```

File: packages/web/components/Player/Section/PlayerHero.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { Box, Flex, HStack, Link, Text, VStack } from '@metafam/ds';
+import { Box, Flex, HStack, Link, Text, VStack, WrapItem } from '@metafam/ds';
```
Comment (dysbulic) [inline packages/web/components/Player/Section/PlayerHero.tsx:1]:
```suggestion
import { Box, Flex, HStack, Link, Text, VStack } from '@metafam/ds';
```

File: packages/web/components/Setup/SetupPronouns.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      input: {
+        pronouns: pronouns ?? '',
+      },
+    });
+
+    if (error) {
+      const errorDetail = 'The octo is sad 😢';
+      toast({
+        title: 'Error',
+        description: `Unable to update Player Pronouns. ${errorDetail}`,
```
Comment (dysbulic) [inline packages/web/components/Setup/SetupPronouns.tsx:42]:
```suggestion
        description: `Unable to update Player pronouns. ${errorDetail}`,
```

File: packages/web/graphql/mutations/player.ts [hunk 6]
```diff
@@ -47,3 +47,13 @@ export const UpdateSkillsMutation = gql`
     }
   }
 `;
+
+export const UpdateProfilePronouns = gql`
+  mutation updateProfilePronouns($playerId: uuid!, $input: player_set_input!) {
+    update_player_by_pk(pk_columns: { id: $playerId }, _set: $input) {
+      id
+      availability_hours
```
Comment (dysbulic) [inline packages/web/graphql/mutations/player.ts:55]:
```suggestion
```

File: packages/web/graphql/mutations/player.ts [hunk 7]
```diff
@@ -47,3 +47,13 @@ export const UpdateSkillsMutation = gql`
...
+      timezone
```
Comment (dysbulic) [inline packages/web/graphql/mutations/player.ts:56]:
```suggestion
```

**Review by dysbulic — CHANGES_REQUESTED** [review]

**Comment by dysbulic** [comment]
¿@jaburx, have you abandoned this code? You're pretty close. If you respond to a couple of my comments, I'll merge this.

**Comment by Udit-takkar** [comment]
Is this issue still open? Can i look into this?

File: packages/web/components/Setup/SetupDone.tsx [hunk 8]
```diff
@@ -7,7 +7,7 @@ import React, { useState } from 'react';
 
 export const SetupDone: React.FC = () => {
   const router = useRouter();
-  const { user } = useUser({ redirectTo: '/' });
```
[the remark this answers is not in the document]
Comment (vidvidvid) [reply packages/web/components/Setup/SetupDone.tsx:10]: this was causing the buggy behaviour when visiting one of the profile setup pages and the user would just get redirected to /players, i fixed it in `packages/web/lib/hooks/index.ts` by renaming the `redirectIfFound` into `redirectIfNotFound` and setting to `false` by default and tweaking some of its behaviour. Now everything works the same, apart from the bug in player setup pages, which is gone.

File: packages/web/components/Setup/SetupPronouns.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    setLoading(true);
+    const { error } = await updatePronouns({
+      playerId: user.id,
+      input: {
+        pronouns: pronouns ?? '',
+      },
+    });
+
+    if (error) {
+      const errorDetail = 'The octo is sad 😢';
```
[the remark this answers is not in the document]
Comment (vidvidvid) [reply packages/web/components/Setup/SetupPronouns.tsx]: Added it.

**Comment by dysbulic** [comment]
This code is based on a version of the profile editing page that doesn't exist anymore. This is in part a result of the time it took to complete the review, which is largely because I was unaware the issues I had raised had been resolved.

I've been trying to figure out how to remove Dependabot from my GitHub activity feed. I turned off notifications in the settings, but I keep getting them. They make it harder to distinguish important updates.

Also, however, the redesign of the profile pages has been ongoing for a almost a month now, so this issue shouldn't be a surprise to anyone.

To me, the correct next step is to put in a request to the design guild for a profile page with pronouns in it. ¿Right, @davort?

**Comment by vidvidvid** [comment]
Hey, the setup wizard steps still exist. They are just accessible only the first time the user logs in and sets up their profile.
Or are you talking about some other non-existent profile editing page?

**Comment by vidvidvid** [comment]
@dysbulic i'm gonna merge the develop branch into it and fix the conflicting files, ok?

**Comment by vidvidvid** [comment]
actually, i would also need to add the pronouns input field to the new profile edit dialog, but this can be done after this pr is merged, so we don't keep bloating it up. what do you think?

**Comment by vidvidvid** [comment]
For some reason the build appears to fail, but when I run `yarn backend:build` locally, it builds it without any errors: `@metafam/backend: [07:47:08] Generate outputs [failed]` is failing it.
I will try to re-run the checks.

**Comment by vidvidvid** [comment]
Closing to re-run

**Comment by vidvidvid** [comment]
Ok, the build has now been completed, but there's still an error when [deploying to cloud run](https://github.com/MetaFam/TheGame/runs/4059106230?check_suite_focus=true#step:6:21) @dysbulic could you check what's up with that?

**Comment by davort** [comment]
> To me, the correct next step is to put in a request to the design guild for a profile page with pronouns in it. ¿Right, @davort?

TBH, I hadn't perceived design to be a blocker in this case. If that's so, I will shift focus onto this. But before I do, is it only that we need to show this data somewhere in the current build for the Profile page? If so, then the data could be inserted by a builder, without having to wait for designs to be fully shaped. 

In any case, it isn't perfectly clear to me if and how design is a blocker in this matter, at this point in time. I would appreciate knowing if it is, and why exactly. Thanks!

**Comment by dysbulic** [comment]
I misread the new interface. I remembered a conversation saying we wanted to keep the profile wizard, but I didn't see a way to access it & assumed the pronouns would need to be displayed and updated in the new UI.

I'm closing this to trigger a rebuild of the cloud instances.

**Comment by dysbulic** [comment]
Rebuild…

**Comment by dysbulic** [comment]
@vidvidvid, just one last quibble. On [the roles page](https://frontend-pr-711-mjhnbmqqna-uk.a.run.app/profile/setup/roles) which precedes pronouns, the submit button is entitled "Professional Profile" rather than "Pronouns".

Also, on the roles page, could you switch `whiteSpace` from `nowrap` to `auto` for the "Make Primary / Secondary" and "Remove Role" buttons when selecting a role? They are overdrawing each other.

2021⁄11⁄02@11:16:37 screenshot profile-setup-roles

File: packages/web/components/Setup/SetupPronouns.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      playerId: user.id,
+      input: {
+        pronouns: pronouns ?? '',
+      },
+    });
+
+    if (error) {
+      toast({
+        title: 'Error',
+        description: `Unable to update Player Pronouns. ${error.message}`,
```
Comment (dysbulic) [inline packages/web/components/Setup/SetupPronouns.tsx:41]: Player could possibly be capitalized as a proper noun, but I think it looks best lowercase.

```suggestion
        description: `Unable to update player pronouns. ${error.message}`,
```

File: packages/web/components/Setup/SetupPronouns.tsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      </MetaHeading>
+      <Input
+        background="dark"
+        placeholder="they/them"
+        value={pronouns ?? ''}
+        onChange={({ target: { value } }) => setPronouns(value)}
+        w="auto"
+      />
+
+      <MetaButton
```
Comment (dysbulic) [inline packages/web/components/Setup/SetupPronouns.tsx:65]: You had this on the rest of them:

```suggestion
      <MetaButton
        disabled={!user}
```

**Review by dysbulic — CHANGES_REQUESTED** [review]
Overall looks good. I had some syntactic suggestions, but the only real blocking issue is the [text of the submit button](https://github.com/MetaFam/TheGame/pull/711#issuecomment-957824504) on the preceding page.

**Comment by alalonde** [comment]
Two requests, now that the edit profile branch is in here:

- only show pronouns section of the player card if there is data set
- add pronoun input field to the edit modal

**Comment by davort** [comment]
> Overall looks good. I had some syntactic suggestions, but the only real blocking issue is the [text of the submit button](https://github.com/MetaFam/TheGame/pull/711#issuecomment-957824504) on the preceding page.

I don't think pronouns as a separate/standalone step in the onboarding flow is optimal.

In principle, we'd want to keep the whole flow as fluid as possible. That said, perhaps we ought to consider "bundling" the choice of pronouns with some other, simpler choices, such as "Display name" or "Website"?

**Comment by vidvidvid** [comment]
WIll take care of @dysbulic 's and @alalonde 's suggestions tomorrow, gonna sleep now. @davort I think your suggestion makes sense and it would be nice to bundle up some of the choices, but I'd prefer to get done with this pr and make another issue for a slight re-work of the setup wizard - that way 1. you get time to make a plan on what to put together and 2. we can finally close this pr that's been open for months, what do you think?

**Comment by davort** [comment]
> WIll take care of @dysbulic 's and @alalonde 's suggestions tomorrow, gonna sleep now. @davort I think your suggestion makes sense and it would be nice to bundle up some of the choices, but I'd prefer to get done with this pr and make another issue for a slight re-work of the setup wizard - that way 1. you get time to make a plan on what to put together and 2. we can finally close this pr that's been open for months, what do you think?

Thanks @vidvidvid - sounds good to me. Let's do like you suggested 👍

File: packages/web/components/Setup/SetupPronouns.tsx (hunk 11 above)
Comment (vidvidvid) [reply packages/web/components/Setup/SetupPronouns.tsx:65]: added

File: packages/web/components/Setup/SetupPronouns.tsx (hunk 10 above)
Comment (vidvidvid) [reply packages/web/components/Setup/SetupPronouns.tsx:41]: fixed

**Comment by vidvidvid** [comment]
Fixed @dysbulic 's suggestions, also slightly improved the buttons from the screenshot you provided:
Screenshot 2021-11-03 at 12 20 06

Also took care of @alalonde 's requests:
- if pronouns weren't selected, hide the section
- added pronoun input field to the edit modal

We can eventually make a follow up with re-organisation of the setup wizard steps.

**Comment by alalonde** [comment]
> 
> I don't think pronouns as a separate/standalone step in the onboarding flow is optimal.
> 

I would go even farther and say that it shouldn't be in the onboarding flow at all. There are probably too many steps already; really only the most critical information should be required during onboarding.

**Review by alalonde — APPROVED** [review]

**Pull request merged**